src/main/java/study/calculator/Numbers.java

```diff
@@ -0,0 +1,52 @@
+package study.calculator;
+
+import study.calculator.pattern.CustomPattern;
+import study.calculator.pattern.DefaultPattern;
+import study.calculator.pattern.PatternSeparator;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+import static study.calculator.Validator.PREFIX_STRATEGY_CONDITION;
+import static study.calculator.Validator.STRING_WHITESPACE;
+
+/**
+ * 숫자를 관리 할 일급 컬렉션
+ */
+public class Numbers {
+
+    private final List<Number> numbers;
+
+    public Numbers() {
+        this(STRING_WHITESPACE);
+    }
+
+    private Numbers(String text) {
+        this(text, new DefaultPattern());
+    }
+
+    protected Numbers(String text, PatternSeparator patternSeparator) {
+        numbers = getExtractNumbers(text, patternSeparator);
+    }
+
+    public static Numbers of(String text) {
+        if(text.startsWith(PREFIX_STRATEGY_CONDITION)) {
+            return new Numbers(text, new CustomPattern());
+        }
+        return new Numbers(text);
+    }
```

Comment:
```suggestion
    public static Numbers of(String text) {
        if (isCustomPatternText(text)) {
            return new Numbers(text, new CustomPattern());
        }
        return new Numbers(text);
    }

    private static boolean isCustomPatternText(String text) {
        return text.startsWith(PREFIX_STRATEGY_CONDITION);
    }
```
조건문을 메서드로 만들고 이름을 부여하면, 가독성이 좋아진다고 생각합니다. 🙂

Reply:
조건에 대해서도 시그니처를 통해 의미있는 코드로 변환 !
좋은 조언 감사드립니다. !!